feat: "Continue Previous Submission" CTA update

olerichter00 opened the pull request:
This PR resolves [ONYX-1030] <!-- eg [PROJECT-XXXX] -->

### Description

This PR updates the "Continue Previous Submission" CTA on the `Sell` screen. If there is a draft submission after the user clicks "Exit & Save" in the submission form, this CTA displays the artist and the artist name and let's the user continue the submission.

| Before | After |
| --- | --- |
|![Simulator Screenshot - iPhone 15 Pro - 2024-06-03 at 15 22 53](https://github.com/artsy/eigen/assets/4691889/0b815415-ee31-4348-b1b8-9f6313f7d270) | ![Simulator Screenshot - iPhone 15 Pro - 2024-06-03 at 15 57 42](https://github.com/artsy/eigen/assets/4691889/e82d7c7d-f4d7-4bf5-8131-5931932d6a2f)|







### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

### Changelog updates

<!-- 📝 Please fill out at least one of these sections. -->
<!-- ⓘ 'User-facing' changes will be published as release notes. -->
<!-- ⌫ Feel free to remove sections that don't apply. -->
<!-- • Write a markdown list or just a single paragraph, but stick to plain text. -->
<!-- 📖 eg. `Enable lotsByFollowedArtists - john` or `Fix phone input misalignment - mary`. -->
<!-- 🤷‍♂️ Replace this entire block with the hashtag `#nochangelog` to avoid updating the changelog. -->
<!-- ⚠️ Prefix with `[NEEDS EXTERNAL QA]` if a change requires external QA -->

#### Cross-platform user-facing changes

- "Continue Previous Submission" CTA design update on Sell screen (behind ff) - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

<!-- end_changelog_updates -->

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md


[ONYX-1030]: https://artsyproduct.atlassian.net/browse/ONYX-1030?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ArtsyOpenSource commented:
<!--
  0 failure: 
  1 warning:  ❌ **An error occu...
  
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 77324608259e3f5bba9dc06b72f70aba86c4021b
</p>

olerichter00 commented on src/app/Scenes/SellWithArtsy/Components/Header.tsx:
```diff
@@ -1,18 +1,34 @@
+  const artist = submission?.artist
 
-  const showContinueSubmission = !!enableSaveAndContinue && !!draft?.submissionID
+  const { trackTappedContinueSubmission } = useSubmitArtworkTracking()
```
This is not yet implemented but I added it here already. Tracking will be implemented in https://github.com/artsy/eigen/pull/10330

olerichter00 commented on src/app/Scenes/SellWithArtsy/SellWithArtsyHome.tsx:
```diff
@@ -158,30 +161,42 @@ const SellWithArtsyHomeContainer = createFragmentContainer(SellWithArtsyHome, {
     me {
       ...SellWithArtsyHome_me
     }
+    submission(id: $submissionID) @include(if: $includeSubmission) {
```
This skips querying for the latest submission if there is no draft.

nickskalkin commented:
@olerichter00 margin between the "Finish previous submission" section and "Sell art from your collection" title seems too big to me, I think it should be reduced

olerichter00 commented:
> @olerichter00 margin between the "Finish previous submission" section and "Sell art from your collection" title seems too big to me, I think it should be reduced

You are right! I've fixed the spacing, updated the screenshot in the description, and added a condition to display the CTA only if the FF is enabled.

nickskalkin reviewed (approved):
LGTM! 👍
